nailgun/entities.py

```diff
@@ -1359,11 +1360,19 @@ def __init__(self, server_config=None, **kwargs):
                 required=True,
             ),
         }
+        super(HostClasses, self).__init__(server_config, **kwargs)
         self._meta = {
-            'api_path': 'api/v2/hosts/:host_id/puppetclass_ids',
             'server_modes': ('sat'),
+            'api_path': '{0}/puppetclass_ids'.format(
+                # pylint:disable=no-member
+                self.host.path('self')
+            )
         }
-        super(HostClasses, self).__init__(server_config, **kwargs)
+
+    def read(self, entity=None, attrs=None, ignore=('hostgroup',)):
+        """Deal with redundant entity fields"""
+        if attrs is None:
+            attrs = self.read_json()
```

Comment:
I think you can drop the assignment to `attrs` completely. The method could be this:

```python
def read(self, entity=None, attrs=None, ignore=('hostgroup',)):
    """Deal with redundant entity fields."""
    return super(HostClasses, self).read(entity, attrs, ignore)
```